pkg/ensurance/collector/nodelocal/net.go

```diff
@@ -0,0 +1,126 @@
+	SentPckps float64
+	// DropIn is the package dropped per second for ingress
+	DropIn float64
+	// DropOut is the package dropped per second for egress
+	DropOut float64
+}
+
+type NetIOCollector struct {
+	netStates map[string]NetTimeStampState
+	ifaces    sets.String
+}
+
+// NewNetIOCollector returns a new Collector exposing kernel/system statistics.
+func NewNetIOCollector(context *NodeLocalContext) (nodeLocalCollector, error) {
+
+	klog.V(2).Infof("NewNetIOCollector")
+
+	return &NetIOCollector{netStates: make(map[string]NetTimeStampState), ifaces: sets.NewString(context.Ifaces...)}, nil
+}
+
+func (n *NetIOCollector) collect() (map[string][]common.TimeSeries, error) {
+	var now = time.Now()
+
+	netIOStats, err := net.IOCounters(true)
+	if err != nil {
+		klog.Errorf("Failed to collect net io resource: %v", err)
+		return map[string][]common.TimeSeries{}, err
+	}
+
+	var netReceiveKiBpsTimeSeries []common.TimeSeries
+	var netSentKiBpsTimeSeries []common.TimeSeries
+	var netReceivePckpsTimeSeries []common.TimeSeries
+	var netSentPckpsTimeSeries []common.TimeSeries
+	var netDropInTimeSeries []common.TimeSeries
+	var netDropOutTimeSeries []common.TimeSeries
+
+	var netStateMaps = make(map[string]NetTimeStampState)
+	for _, v := range netIOStats {
+		if v.Name == "" {
+			continue
+		}
+
+		if !n.ifaces.Has(v.Name) {
+			continue
+		}
+
+		netStateMaps[v.Name] = NetTimeStampState{stat: v, timestamp: now}
+		if vv, ok := n.netStates[v.Name]; ok {
+			netIOUsage := calculateNetIO(vv, netStateMaps[v.Name])
+			netReceiveKiBpsTimeSeries = append(netReceiveKiBpsTimeSeries, common.TimeSeries{Labels: []common.Label{{Name: "NetInterface", Value: v.Name}}, Samples: []common.Sample{{Value: netIOUsage.ReceiveKibps, Timestamp: now.Unix()}}})
+			netSentKiBpsTimeSeries = append(netSentKiBpsTimeSeries, common.TimeSeries{Labels: []common.Label{{Name: "NetInterface", Value: v.Name}}, Samples: []common.Sample{{Value: netIOUsage.SentKibps, Timestamp: now.Unix()}}})
+			netReceivePckpsTimeSeries = append(netReceivePckpsTimeSeries, common.TimeSeries{Labels: []common.Label{{Name: "NetInterface", Value: v.Name}}, Samples: []common.Sample{{Value: netIOUsage.ReceivePckps, Timestamp: now.Unix()}}})
+			netSentPckpsTimeSeries = append(netSentPckpsTimeSeries, common.TimeSeries{Labels: []common.Label{{Name: "NetInterface", Value: v.Name}}, Samples: []common.Sample{{Value: netIOUsage.SentPckps, Timestamp: now.Unix()}}})
+			netDropInTimeSeries = append(netDropInTimeSeries, common.TimeSeries{Labels: []common.Label{{Name: "NetInterface", Value: v.Name}}, Samples: []common.Sample{{Value: netIOUsage.DropIn, Timestamp: now.Unix()}}})
+			netDropOutTimeSeries = append(netDropOutTimeSeries, common.TimeSeries{Labels: []common.Label{{Name: "NetInterface", Value: v.Name}}, Samples: []common.Sample{{Value: netIOUsage.DropOut, Timestamp: now.Unix()}}})
+		}
+	}
+
+	n.netStates = netStateMaps
+
+	var storeMaps = make(map[string][]common.TimeSeries, 0)
```

Comment:
stateMap not storeMaps? map is a collection, should be in singular.

Reply:
done~